internal/transport/grpchttp2/http2bridge.go

```diff
@@ -0,0 +1,219 @@
+// Framer implementation to be able to work with the grpchttp2.Framer interface.
+// This type exists to give an opt-out feature for the new framer and it is
+// eventually going to be removed.
+type FramerBridge struct {
+	framer *http2.Framer  // the underlying http2.Framer implementation to perform reads and writes.
+	pool   mem.BufferPool // a pool to reuse buffers when reading.
+}
+
+// NewFramerBridge creates a new framer by taking a writer and a reader,
+// alongside the maxHeaderListSize for the maximum size of the headers table.
+// The underlying framer uses the SetReuseFrames feature to avoid extra
+// allocations.
+func NewFramerBridge(w io.Writer, r io.Reader, maxHeaderListSize uint32) *FramerBridge {
+	fr := &FramerBridge{
+		framer: http2.NewFramer(w, r),
+		pool:   mem.DefaultBufferPool(),
+	}
+
+	fr.framer.SetReuseFrames()
+	fr.framer.MaxHeaderListSize = maxHeaderListSize
+	fr.framer.ReadMetaHeaders = hpack.NewDecoder(initHeaderTableSize, nil)
+
+	return fr
+}
+
+// ReadFrame reads a frame from the underlying http2.Framer and returns a
+// Frame defined in the grpchttp2 package.
+func (fr *FramerBridge) ReadFrame() (Frame, error) {
+	f, err := fr.framer.ReadFrame()
+	if err != nil {
+		return nil, err
+	}
+
+	h := f.Header()
+	hdr := &FrameHeader{
+		Size:     h.Length,
+		Type:     FrameType(h.Type),
+		Flags:    Flag(h.Flags),
+		StreamID: h.StreamID,
+	}
+
+	switch f := f.(type) {
+	case *http2.DataFrame:
+		buf := fr.pool.Get(int(hdr.Size))
+		copy(buf, f.Data())
+		df := &DataFrame{
+			hdr:  hdr,
+			Data: buf,
+		}
+		df.free = func() { fr.pool.Put(buf) }
+		return df, nil
+	case *http2.HeadersFrame:
+		buf := fr.pool.Get(int(hdr.Size))
+		copy(buf, f.HeaderBlockFragment())
+		hf := &HeadersFrame{
+			hdr:      hdr,
+			HdrBlock: buf,
+		}
+		hf.free = func() { fr.pool.Put(buf) }
+		return hf, nil
+	case *http2.RSTStreamFrame:
+		return &RSTStreamFrame{
+			hdr:  hdr,
+			Code: ErrCode(f.ErrCode),
+		}, nil
+	case *http2.SettingsFrame:
+		buf := make([]Setting, 0, f.NumSettings())
+		f.ForeachSetting(func(s http2.Setting) error {
+			buf = append(buf, Setting{
+				ID:    SettingID(s.ID),
+				Value: s.Val,
+			})
+			return nil
+		})
+		return &SettingsFrame{
+			hdr:      hdr,
+			Settings: buf,
+		}, nil
+	case *http2.PingFrame:
+		buf := fr.pool.Get(int(hdr.Size))
+		copy(buf, f.Data[:])
+		pf := &PingFrame{
+			hdr:  hdr,
+			Data: buf,
+		}
+		pf.free = func() { fr.pool.Put(buf) }
+		return pf, nil
+	case *http2.GoAwayFrame:
+		// Size of the frame minus the code and lastStreamID
+		buf := fr.pool.Get(int(hdr.Size) - 8)
+		copy(buf, f.DebugData())
+		gf := &GoAwayFrame{
+			hdr:          hdr,
+			LastStreamID: f.LastStreamID,
+			Code:         ErrCode(f.ErrCode),
+			DebugData:    buf,
+		}
+		gf.free = func() { fr.pool.Put(buf) }
+		return gf, nil
+	case *http2.WindowUpdateFrame:
+		return &WindowUpdateFrame{
+			hdr: hdr,
+			Inc: f.Increment,
+		}, nil
+	case *http2.ContinuationFrame:
```

Comment:
Hmm ... I don't see any handling of this frame type in our current code. Do you happen to know how they are handled currently? Thanks.